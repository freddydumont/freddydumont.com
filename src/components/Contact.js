/** @jsx jsx **/
import { jsx } from 'theme-ui';
import Section from './Section';
import {
  Button,
  Label,
  Flex,
  Box,
  Input,
  Textarea,
} from '@theme-ui/components';

const Contact = () => (
  <Section htmlId="contact" withContainer>
    <Section.Title>Contact</Section.Title>
    <Section.Body>Use the form below to send me an email.</Section.Body>
    <form
      sx={{ width: 'full' }}
      data-netlify="true"
      method="post"
      name="contact"
    >
      <Input type="hidden" name="form-name" value="contact" />
      <Flex variant="flex_form_element">
        <Box
          variant="form_box"
          mb={[12, 12, 0]}
          sx={{
            width: ['full', 'full', '1/2'],
          }}
        >
          <Label htmlFor="first-name">First Name*</Label>
          <Input type="text" name="first-name" id="first-name" required />
        </Box>
        <Box
          variant="form_box"
          sx={{
            width: ['full', 'full', '1/2'],
          }}
        >
          <Label htmlFor="last-name">Last Name</Label>
          <Input type="text" name="last-name" id="last-name" />
        </Box>
      </Flex>
      <Flex variant="flex_form_element">
        <Box variant="form_box">
          <Label htmlFor="email">Email address*</Label>
          <Input type="email" name="email" id="email" required />
        </Box>
      </Flex>
      <Flex variant="flex_form_element">
        <Box variant="form_box">
          <Label htmlFor="message">Message*</Label>
          <Textarea rows="8" name="message" id="message" required />
        </Box>
      </Flex>
      <Flex
        variant="flex_form_element"
        sx={{
          justifyContent: 'flex-end',
        }}
      >
        <Box
          variant="form_box"
          sx={{
            maxWidth: ['initial', 'initial', 'xs'],
          }}
        >
          <Button
            variant="gray"
            data-testid="button"
            type="submit"
            sx={{
              ':hover': {
                boxShadow: '2xl',
              },
            }}
          >
            Send
          </Button>
        </Box>
      </Flex>
    </form>
  </Section>
);

export default Contact;
