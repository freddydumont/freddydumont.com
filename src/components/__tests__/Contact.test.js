import React from 'react';
import { render } from '../../../test/test-utils';
import Contact from '../Contact';

describe('Contact form', () => {
  const { getByLabelText, getByText, getByTestId } = render(<Contact />);
  const title = getByText('Contact');
  const button = getByTestId('button');
  const firstName = getByLabelText('First Name*');
  const lastName = getByLabelText('Last Name');
  const email = getByLabelText('Email address*');
  const message = getByLabelText('Message*');

  it('should display a title', () => {
    expect(title).toBeInTheDocument();
  });

  it('should have a required first name field', () => {
    expect(firstName).toBeRequired();
  });

  it('should have an optional last name field', () => {
    expect(lastName).not.toBeRequired();
  });

  it('should have a required email field', () => {
    expect(email).toBeRequired();
  });

  it('should have a required message field', () => {
    expect(message).toBeRequired();
  });

  it('should have a send button', () => {
    expect(button).toHaveTextContent('Send');
  });

  it('should match snapshot', () => {
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });
});
