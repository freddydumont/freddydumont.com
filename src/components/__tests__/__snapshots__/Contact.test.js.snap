// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact form should match snapshot 1`] = `
.emotion-23 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  height: full;
  color: gray.2;
  margin-top: 16px;
}

@media screen and (min-width:40em) {
  .emotion-23 {
    margin-top: 32px;
  }
}

.emotion-22 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  max-width: container;
  margin-left: auto;
  margin-right: auto;
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 32px;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 12px;
}

.emotion-21 {
  width: full;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 12px;
  width: full;
}

@media screen and (min-width:40em) {
  .emotion-5 {
    margin-bottom: 12px;
  }
}

@media screen and (min-width:52em) {
  .emotion-5 {
    margin-bottom: 0;
  }
}

@media screen and (min-width:40em) {
  .emotion-5 {
    width: full;
  }
}

@media screen and (min-width:52em) {
  .emotion-5 {
    width: 1/2;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: full;
}

@media screen and (min-width:40em) {
  .emotion-8 {
    width: full;
  }
}

@media screen and (min-width:52em) {
  .emotion-8 {
    width: 1/2;
  }
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
}

.emotion-20 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-19 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: initial;
}

@media screen and (min-width:40em) {
  .emotion-19 {
    max-width: initial;
  }
}

@media screen and (min-width:52em) {
  .emotion-19 {
    max-width: xs;
  }
}

.emotion-18 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: primary;
  border: 0;
  border-radius: 4px;
}

.emotion-18:hover {
  box-shadow: 2xl;
}

<div>
  <section
    class="emotion-23"
    id="contact"
  >
    <div
      class="emotion-22"
    >
      <h1
        class="emotion-0"
      >
        Contact
      </h1>
      <p
        class="emotion-1"
      >
        Use the form below to send me an email.
      </p>
      <form
        class="emotion-21"
        data-netlify="true"
        method="post"
        name="contact"
      >
        <input
          class="emotion-2"
          name="form-name"
          type="hidden"
          value="contact"
        />
        <div
          class="emotion-9"
        >
          <div
            class="emotion-5"
          >
            <label
              class="emotion-3"
              for="first-name"
            >
              First Name*
            </label>
            <input
              class="emotion-2"
              id="first-name"
              name="first-name"
              required=""
              type="text"
            />
          </div>
          <div
            class="emotion-8"
          >
            <label
              class="emotion-3"
              for="last-name"
            >
              Last Name
            </label>
            <input
              class="emotion-2"
              id="last-name"
              name="last-name"
              type="text"
            />
          </div>
        </div>
        <div
          class="emotion-9"
        >
          <div
            class="emotion-12"
          >
            <label
              class="emotion-3"
              for="email"
            >
              Email address*
            </label>
            <input
              class="emotion-2"
              id="email"
              name="email"
              required=""
              type="email"
            />
          </div>
        </div>
        <div
          class="emotion-9"
        >
          <div
            class="emotion-12"
          >
            <label
              class="emotion-3"
              for="message"
            >
              Message*
            </label>
            <textarea
              class="emotion-2"
              id="message"
              name="message"
              required=""
              rows="8"
            />
          </div>
        </div>
        <div
          class="emotion-20"
        >
          <div
            class="emotion-19"
          >
            <button
              class="emotion-18"
              data-testid="button"
              type="submit"
            >
              Send
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>
</div>
`;
